composer/core/types.py

```diff
@@ -141,6 +142,58 @@ def __len__(self) -> int:
         ...
 
 
+@dataclass
+class Evaluator:
+    """Wrapper for a dataloader to include metrics that apply to a specific
+    dataset.
+
+    Attributes:
+        label (str): Name of the Evaluator
+        dataloader (Union[DataLoader, DataSpec]): Dataloader/DataSpec for evaluation data
+        metrics (Optional[Metrics]): Metrics to use for the dataset.
+        validate_every_n_batches (Optional[int]): Compute metrics on evaluation data every N batches.
+             Set to -1 to never validate on a batchwise frequency. (default: ``-1``)
+        validate_every_n_epochs (Optional[int]): Compute metrics on evaluation data every N epochs.
+            Set to -1 to never validate on a epochwise frequency. (default: ``1``)
+        metric_names: (Optional[List[str]]): List of string names for desired metrics in an Evaluator. If specified,
+            the trainer will look through the compatible metrics for a model and populate the metrics field
+            with torchmetrics with names appearing in metric_names.
+        eval_subset_num_batches (int, optional): If specified, evaluate on this many batches.
+            This parameter has no effect if it is greater than ``len(eval_dataloader)``.
+            If None (the default), then the entire dataloader will be iterated over.
+        device_transforms ((Batch) -> Batch, optional): Function that is called by the trainer to modify the batch
+            once it has been moved onto the device. For example, this function can be used for GPU-based normalization.
+            It can modify the batch in-place, and it should return the modified batch. If omitted, the batch is not
+            modified.
+    """
+
+    label: str
+    dataloader: Union[DataLoader, DataSpec]
+    metrics: Metrics = None
+    validate_every_n_epochs: int = 1
+    validate_every_n_batches: int = -1
+    metric_names: Optional[Sequence[str]] = None
+    eval_subset_num_batches: Optional[int] = None
```

Comment:
remove this and use the `eval_subset_num_batches` in the Trainer `init` signature